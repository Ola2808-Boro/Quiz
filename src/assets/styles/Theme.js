export const theme = {
  colors: {
    backgroundColor: "#F5F7FB",
    text: "#293264",
    buttonColor: "#F5F7FB",
    circleBottom: "#DEEBF8",
    circleUp: "#FFFAD1",
    buttonBg: "#4D5B9E",
    borderAnswer: "#4D5B9E",
  },

  fontSize: {
    huge: "70px",
    xlgrand: "44px",
    lgrand: "40px",
    grand: "37px",
    xxxxl: "34px",
    xxxl: "33px",
    xxl: "30 px",
    xl: "26px",
    mediumxl: "24px",
    l: "22px",
    ml: "18px",
    m: "14px",
    s: "10px",
  },
};
