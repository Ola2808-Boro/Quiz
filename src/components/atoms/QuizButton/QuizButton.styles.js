import styled from "styled-components";
import { device } from "helpers/breakpoints";

export const StyledButton = styled.button`
  margin-top: 1rem;
  font-size: ${({ theme }) => theme.fontSize.l};
  background-color: ${({ theme }) => theme.colors.buttonBg};
  letter-spacing: 1px;
  border-radius: 10px;
  color: white;
  border: none;
  padding: 10px 20px;
  width: 30%;
  margin-bottom: 1rem;
  z-index: 1;
  @media ${device.mobileS} {
    width: 70%;
  }

  @media ${device.tablet} {
    width: 50%;

    font-size: ${({ theme }) => theme.fontSize.xl};
  }
  @media ${device.laptop} {
    width: 50%;

    font-size: ${({ theme }) => theme.fontSize.xxl};
  }
  @media ${device.laptopL} {
    width: 30%;
  }
  @media ${device.desktop} {
    width: 20%;
    font-size: ${({ theme }) => theme.fontSize.grand};
  }
`;
