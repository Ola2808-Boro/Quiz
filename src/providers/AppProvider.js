import React from 'react';
import { GlobalStyle } from 'assets/styles/GlobalsStyle'
import { ThemeProvider } from 'styled-components';
import 'assets/styles/fonts.css'
import {theme} from 'assets/styles/Theme'


const AppProviders = ({ children }) => {
    return (
 
          <ThemeProvider theme={theme}>      
                <GlobalStyle />
                {children}
          </ThemeProvider>
 
    );
  };
  
  export default AppProviders;