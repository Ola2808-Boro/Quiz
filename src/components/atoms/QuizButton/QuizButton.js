import React,{useContext} from 'react';
import { StyledButton } from './QuizButton.styles';
import { quizContext } from 'providers/QuizProvider';
const QuizButton = () => {

    const {check,handleButtonClick}= useContext(quizContext)
    
    return (
       <StyledButton onClick={handleButtonClick}>
       {check ? 'Try again' : 'Check answers'}
       </StyledButton>
    );
};


export default QuizButton;