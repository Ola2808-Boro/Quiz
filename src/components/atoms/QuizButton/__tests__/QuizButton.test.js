import React from "react";
import "jest-styled-components";
import renderer from "react-test-renderer";
import AppProvider from "providers/AppProvider";
import { StyledButton } from "components/atoms/QuizButton/QuizButton.styles";
import QuizButton from "../QuizButton";
import { render, screen } from "@testing-library/react";

describe("Test Quiz Button", () => {
  it("Test default style", () => {
    const tree = renderer
      .create(
        <AppProvider>
          <StyledButton />
        </AppProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Test render context", () => {
    render(
      <AppProvider>
        <QuizButton />
      </AppProvider>
    );
    expect(
      screen.getByRole("button", { name: "Check answers" })
    ).toBeInTheDocument();
  });
});
