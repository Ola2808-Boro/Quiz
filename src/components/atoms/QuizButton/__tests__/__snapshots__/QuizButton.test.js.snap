// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Quiz Button Test default style 1`] = `
<button
  className="cGdJYJ"
/>
`;
